packages/vda-did/src/blockchain/blockchainApi.ts

```diff
@@ -0,0 +1,182 @@
+import { getVeridaSignWithNonce, interpretIdentifier } from "./helpers"
+import { VdaDidConfigurationOptions } from "../interfaces"
+import {getContractInfoForNetwork} from "./config"
+import { getVeridaContract, VeridaContract } from "@verida/web3"
+import { ethers } from "ethers"
+
+export interface LookupResponse {
+    didController: string
+    endpoints: string[]
+}
+
+export default class BlockchainApi {
+
+    private options: VdaDidConfigurationOptions
+
+    private didAddress : string
+
+    private vdaWeb3Client : VeridaContract;
+
+    constructor(options: VdaDidConfigurationOptions) {
+        this.options = options
+
+        const { address, publicKey, network } = interpretIdentifier(options.identifier)
+
+        this.didAddress = address;
+
+        const net = network || options.chainNameOrId
+        const contractInfo = getContractInfoForNetwork(net);
+
+        this.vdaWeb3Client = getVeridaContract(
+            options.callType, 
+            {...contractInfo,
+            ...options.web3Options});
+    }
+
+    /**
+     * Get a nonce from DIDRegistry contract
+     * @returns nonce of DID
+     */
+    public async nonceFN() {
+        const response = await this.vdaWeb3Client.nonce(this.didAddress);
+        if (response.data === undefined) {
+            throw new Error('Error in getting nonce');
+        }
+        return response.data;
+    }
+
+    /**
+     * Get a controller & Endpoints' array of a DID address fro blockchain
+     * @returns Controller & Endpoints array
+     */
+    public async lookup(): Promise<LookupResponse> {
```

Comment:
This needs to lookup **any** did document. It could just use `vda-did-resolver`?

`lookup(did: string): DIDDocument`

Reply:
Possible. will update